### PR TITLE
Small changes & fixes

### PR DESCRIPTION
**Pull request by rgerd**
- Moved the morton code calculation logic out into its own shader.
- Improved acceleration structure visualization so you can now see how deeply the ray traversed in the BVH tree to hit each rendered triangle. Try it out!
- Minor semantic fixes on comments/a function/a size calculation.

as visualization

### CODE REVIEW

**Comment by pkristof** [comment]
Yeah, the visualization now looks more as expected. 

Can you add an example image of the visualization and description of color coding to the readme? https://github.com/Microsoft/DirectX-Graphics-Samples/blob/develop-dxr-rs5/Libraries/D3D12RaytracingFallback/developerguide.md#acceleration-structure-visualization

File: Libraries/D3D12RaytracingFallback/src/CalculateMortonCodesForAABBs.hlsl [hunk 1]
```diff
@@ -26,3 +26,5 @@ float3 GetCentroid(uint elementIndex)
 {
     return ReadBottomLevelBoundingBox(elementIndex).center;
 }
+
+#include "CalculateMortonCodes.hlsli"
```
Comment (pkristof) [inline Libraries/D3D12RaytracingFallback/src/CalculateMortonCodesForAABBs.hlsl]: The includes should be with the rest of the includes at the top of the file.

Comment (rgerd) [reply Libraries/D3D12RaytracingFallback/src/CalculateMortonCodesForAABBs.hlsl]: I'm following a pattern (seen in the [top-level compute AABB pass](https://github.com/Microsoft/DirectX-Graphics-Samples/blob/develop-dxr-rs5/Libraries/D3D12RaytracingFallback/src/TopLevelComputeAABBs.hlsl), for example) where the shader is defining helper functions that the [main logic will eventually use](https://github.com/Microsoft/DirectX-Graphics-Samples/pull/435/files/eec10f868bd796311c7ad3fab2dc0a710552cef8#diff-5d48fb565cbf48d0c7d782e5ec8eac59R114) "below" it.

File: Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl [hunk 2]
```diff
@@ -59,7 +59,7 @@ void FormTreelet(in uint groupThreadId)
             {
                 uint treeletNodeIndex = treeletToReorder[i];
                 // Leaf nodes can't be split so skip these
-                if (!IsLeaf(treeletNodeIndex))
+                if (!IsLeafIndex(treeletNodeIndex))
```
Comment (pkristof) [inline Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl]: IMO IsLeaf is better as it's easier to read and comprehend. I don't see it being necessary to stress the index part. I could see an argument for IsLeafIndex() if there was IsLeaf() for an object reference/pointer, but that's not the case and even then IsLeaf should be clear enough.

Comment (pkristof) [reply Libraries/D3D12RaytracingFallback/src/TreeletReorder.hlsl]: That makes sense.

File: Libraries/D3D12RaytracingFallback/src/CalculateMortonCodesForAABBs.hlsl (hunk 1 above)
Comment (pkristof) [reply Libraries/D3D12RaytracingFallback/src/CalculateMortonCodesForAABBs.hlsl]: I find it weird, but I'm OK with if it's already a trend.

Comment (clandrew) [reply Libraries/D3D12RaytracingFallback/src/CalculateMortonCodesForAABBs.hlsl]: +1 this pattern is weird

File: Libraries/D3D12RaytracingFallback/developerguide.md [hunk 3]
```diff
@@ -209,14 +209,17 @@ The Fallback Layer natively works with PIX. However, PIX does not have support f
 When using the debug version of the Fallback Layer library, the Fallback layer has limited validation that will catch unsupported cases and output the cause of failure to the debugger. 
 
 ### Acceleration structure visualization
-It can be helpful to visualize the acceleration structure, particularly when you make changes to the structure logic in the library. To do this, enable ENABLE_ACCELERATION_STRUCTURE_VISUALIZATION in [FallbackDebug.h](src/FallbackDebug.h) and set FallbackLayer::m_levelToVisualize in [FallbackLayer.h](src/FallbackLayer.h) to a tree level you want to visualize.
+It can be helpful to visualize the acceleration structure, particularly when you make changes to the structure logic in the library. To do this, enable ENABLE_ACCELERATION_STRUCTURE_VISUALIZATION in [FallbackDebug.h](src/FallbackDebug.h) and in [ModelViewerRayTracing.h](../../Samples/Desktop/D3D12Raytracing/src/D3D12RaytracingMiniEngineSample/ModelViewerRayTracing.h).
```
Comment (clandrew) [inline Libraries/D3D12RaytracingFallback/developerguide.md]: Does it really need to be defined in two places?

**Comment by clandrew** [comment]
Could you consider renaming this PR to have a more descriptive title? Like "Refactor Morton code calculation" or something like that

File: Libraries/D3D12RaytracingFallback/developerguide.md (hunk 3 above)
Comment (rgerd) [reply Libraries/D3D12RaytracingFallback/developerguide.md]: ....yyyes?

I think?

**Pull request merged**